datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -1590,6 +1590,13 @@ fn build_statistics_expr(
                 )),
             ))
         }
+        Operator::NotLikeMatch => {
+            build_not_like_match(expr_builder).ok_or_else(|| {
+                plan_datafusion_err!(
+                    "The NOT LIKE expression with wildcards is only supported at the end of the pattern"
```

Comment:
Just a general note, not necessarily for this PR: I do think we should somehow make these errors (also for the existing LIKE case) a bit better if there is any other reason they might be rejected? Or if this is the only reason, a more generic error might be better? I realize these errors likely never surface to users but I think it'd be unfortunate to see an error like this one when the actual reason it wasn't supported has nothing to do with it.

Reply:
Agree. Now it return `Result<Arc<dyn PhysicalExpr>>`